src/JsonMapper.php

```
@@ -287,22 +343,294 @@ protected function getMappedValue(
+            if ($type->getGroupName() == 'anyOf') {
+                break; // break if its anyOf, and we already have mapped its value
+            }
+        }
+        if (!$mappedWith) {
+            throw new JsonMapperException(
+                'Unable to map AnyOf ' .
+                TypeCombination::generateTypeString($type) .
+                ' on: ' . json_encode($json)
+            );
+        }
+        return $mappedObject;
+    }
+
+    /**
+     * Checks types against the value.
+     *
+     * @param mixed         $value          param's value
+     * @param string        $type           type defined in param's typehint
+     * @param string[]|null $factoryMethods Callable factory methods for property
+     * @param string        $namespace      Namespace of the class
+     * @param string        $className      Class refrencing the factory methods
+     *
+     * @return array array(bool $matched, ?string $method) $matched represents if
+     *               Type matched with value, $method represents the selected
+     *               factory method (if any)
+     * @throws ReflectionException
+     * @throws JsonMapperException
+     */
+    private function _isValueOfType(
+        $value,
+        $type,
+        $factoryMethods,
+        $namespace,
+        $className
+    ) {
+        if (isset($factoryMethods)) {
+            $methodFound = false;
+            foreach ($factoryMethods as $method) {
+                if (isset($method) && explode(' ', $method)[1] == $type) {
+                    $methodFound = true;
+                    if (version_compare(phpversion(), '7.0', '<')) {
+                        // if php version is less than 7.0
+                        $this->_callFactoryMethod($method, $value, $className);
+                    } else {
+                        try {
+                            $this->_callFactoryMethod($method, $value, $className);
+                        } catch (\Throwable $e){
```

I don't understand this part of the code; you are catching _all_ exceptions and then ignoring the exception to continue the loop. What is this for-loop doing and did you consider a better code than using exceptions in the main flow of your code (main flow meaning the normal running of the code)?

---

This loop is checking for each deserializer function seprately if any one can be used to deserailze the given value, and yes there is no other way to check if a deserializer function is throwing an exception or running as expected, since it will be user defined